tests/trainers/test_classification.py

```diff
@@ -110,6 +111,21 @@ def model_kwargs(self) -> Dict[str, Any]:
             "weights": None,
         }
 
+    @pytest.fixture(params=[*ResNet50_Weights])
+    def weights(self, request: SubRequest) -> WeightsEnum:
+        return request.param
+
+    @pytest.fixture
+    def mocked_weights(
```

Comment:
If you use the same name as the below fixture, it's just going to be redefined and this won't be used. You'll want to have a single fixture that iterates over all enums. I can hack on this later if you can't figure it out.

Reply:
I think I am doing something wrong, at least cannot figure out the failing tests. I have also run this, which works with the new load_state_dict for all models except the seco weights, because the module key is not in the state dict so looking into that.

```
for model in torchgeo.models.list_models():
    for weight in torchgeo.models.get_model_weights(model):
        try:
            task = ClassificationTask(
                model=model,
                weights=weight,
                num_classes=10,
                in_channels=13,
                loss="ce"
            )
        except:
            print(model, weight)
```

So I think I need your help.

Reply:
Fixed. You were extremely close. The only thing missing was changing the "model"/"in_channels" from `model_kwargs` to match `mocked_weights`. Also simplified things a bit, the `WeightsEnum` contains the model name. Also added tests to make sure these work with the real weights (only run on release branch).

Reply:
There are still issues with the SeCo weights as you noted. I'm surprised that the first layer seems to be missing from the pretrained model, this is usually the most important layer. Can you look into this? Worst case scenario, we can copy the first layer from an ImageNet pretrained model (will obvs need to test this to make sure it works).

Also, can you copy these tests to the other trainers?